### PR TITLE
Dense implementation 

### PR DESCRIPTION
**Pull request by akshaychawla**
First implementation of Dense (fully connected layer)

1. defined forward, backward, return_weights, return_grads
2. UNTESTED CODE. DO NOT EXPECT ANYTHING

ignore vim .swp files

### CODE REVIEW

**Comment by akshaychawla** [comment]
@priyamtejaswin check the backward pass. I'm assuming input is in the form (batchsize, features). also abtract class should also have a return_gradients function or something to return gradients for backward pass.

File: layers/dense.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+from abstract_layer import AbstractLayer 
+import numpy as np
+from copy import deepcopy
+
+class Dense(AbstractLayer):
+    """ Fully connected layer """ 
+
+    def __init__(self, n_in, n_out):
```
Comment (priyamtejaswin) [inline layers/dense.py:8]: Should have a `self.name` property? It'll be easier to manager layers.

File: layers/dense.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        self.output_grad    = deepcopy(current_error)
+        self.input_grad     = np.dot(self.output_grad, self.weights) 
+        self.weights_grad   = np.dot(self.output_grad, self.input) 
+        self.bias_grad      = self.output_grad * 1.0 
+        return self.input_grad
+
+    def return_weights(self):
+        return [self.weights, self.bias] 
+    
+    def return_grads(self):
```
Comment (priyamtejaswin) [inline layers/dense.py:34]: Thanks for pointing this out. Can you add it to the abc layers?

**Review by priyamtejaswin — CHANGES_REQUESTED** [review]

**Comment by priyamtejaswin** [comment]
Also, update the import in the init file.
[edit]Did this.

**Comment by priyamtejaswin** [comment]
I'll check the backward pass.

**Comment by priyamtejaswin** [comment]
@akshaychawla  i've configured a status check on master merges so that we don't accidentally overwrite.

File: layers/dense.py (hunk 2 above)
Comment (priyamtejaswin) [reply layers/dense.py:34]: Did this.

**Comment by priyamtejaswin** [comment]
Closing PR till we resolve the backward_pass in Dense.

**Pull request closed without merging**